bin/prover-client/src/prover.rs

```diff
@@ -51,13 +57,19 @@ impl ProverState {
     fn set_to_proved(
         &mut self,
         task_id: Uuid,
-        proof: Result<Proof, anyhow::Error>,
+        proof: Result<ProofWithVkey, anyhow::Error>,
     ) -> Option<ProvingTaskState> {
         match proof {
-            Ok(p) => self
-                .tasks_status
-                .insert(task_id, ProvingTaskState::Proved(p)),
-            Err(e) => self.tasks_status.insert(task_id, ProvingTaskState::Err(e)),
+            Ok(p) => {
+                info!("Completed proving task {:?}", task_id);
+                self.tasks_status
+                    .insert(task_id, ProvingTaskState::Proved(p))
+            }
+            Err(e) => {
+                error!("Error proving {:?} {:?}", task_id, e);
+                self.tasks_status
+                    .insert(task_id, ProvingTaskState::Err(e.to_string()))
+            }
```

Comment:
Looking at the body, looks like the function should be named something like `set_status`